### PR TITLE
remove support for int8 branches in NANO

### PR DESCRIPTION
**Pull request by vlimant**
#### PR description:

branches of type int8_t / char_t n NANO are creating various numerical issues for user reading nano root files.
#43699 

#### PR validation:

the nano matrix was ran to completion

**This PR might be updated to remove uint8_t**

### CODE REVIEW

**Comment by vlimant** [comment]
enable nano

**Comment by vlimant** [comment]
there's an issue with the MUO custom nano from #43065 that needs to be resolved

**Comment by vlimant** [comment]
@battibass : please take a look at my changes and let me know (PR still WIP)

**Comment by vlimant** [comment]
please test

**Comment by battibass** [comment]
> @battibass : please take a look at my changes and let me know (PR still WIP)

The size increase is OK for Muon DPGs.
During development we simply used `int8` as we saw it was an option.

We noticed the issue, but ignored it as, given the way our analysis programs were written, it did not generate problems (at least no problems that I am aware of).

**Comment by vlimant** [comment]
+1

File: PhysicsTools/NanoAOD/python/photons_cff.py [hunk 1]
```diff
@@ -232,7 +232,7 @@ def make_bitmapVID_docstring(id_modules_working_points_pset):
         isScEtaEB = Var("abs(superCluster().eta()) < 1.4442",bool,doc="is supercluster eta within barrel acceptance"),
         isScEtaEE = Var("abs(superCluster().eta()) > 1.566 && abs(superCluster().eta()) < 2.5",bool,doc="is supercluster eta within endcap acceptance"),
         seedGain = Var("userInt('seedGain')","uint8",doc="Gain of the seed crystal"),
-        seediEtaOriX = Var("superCluster().seedCrysIEtaOrIx","int8",doc="iEta or iX of seed crystal. iEta is barrel-only, iX is endcap-only. iEta runs from -85 to +85, with no crystal at iEta=0. iX runs from 1 to 100."),
+        seediEtaOriX = Var("superCluster().seedCrysIEtaOrIx","int16",doc="iEta or iX of seed crystal. iEta is barrel-only, iX is endcap-only. iEta runs from -85 to +85, with no crystal at iEta=0. iX runs from 1 to 100."),
         seediPhiOriY = Var("superCluster().seedCrysIPhiOrIy",int,doc="iPhi or iY of seed crystal. iPhi is barrel-only, iY is endcap-only. iPhi runs from 1 to 360. iY runs from 1 to 100."),
```
Comment (hqucms) [inline PhysicsTools/NanoAOD/python/photons_cff.py:236]: @vlimant Should we also change `seediPhiOriY` to `int16` to be consistent?

Comment (vlimant) [reply PhysicsTools/NanoAOD/python/photons_cff.py:236]: would make sense indeed

**Comment by vlimant** [comment]
please test

**Comment by vlimant** [comment]
+1

**Comment by rappoccio** [comment]
+1

**Pull request merged**